### PR TITLE
Isolate offloaded indexing readme

### PR DESCRIPTION
**Pull request by caseylitton**

### CODE REVIEW

File: README-offloaded-indexing.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        Prod: --profile-name production --candidate
+
+    # Check environment vars
+    $ cat /etc/environment
+        * ENCD_APP_WORKERS=12
+        * ENCD_INDEX_PROCS=8
+        * ENCD_REMOTE_INDEXING=true
+
+    # ssh on and stop apache after the instance restart
+    $ sudo service apache2 stop
```
Comment (caseylitton) [inline README-offloaded-indexing.md:24]: This is unfortunate but by design the head node needs to be built first.  And will start indexing if the offloaded indexing node doesn't not exist.

File: README-offloaded-indexing.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    set queue_worker_processes = 8
+    set remote_indexing = true
+    set remote_indexing_threshold = 10001
+
+    # Check apache
+    App workers should be 12
+    indexer, vis, and region should exist
+
+    # Indexer node: 8 app workers, 16 index workers
+    $ export PG_IP=172.31.30.133
```
Comment (keenangraham) [inline README-offloaded-indexing.md:37]: Is this the frontend?

File: README-offloaded-indexing.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    # Normal elasticsearch cluster
+    $ bin/deploy -b "$GIT_REL" --cluster-name "$ENCD_REL" --es-wait --dry-run
+        RC:   --release-candidate
+        Test: --test
+        Prod: --profile-name production --candidate
+
+    # Head frontend node: 12 app workers, 8 index workers
+    $ export ES_IP=172.31.25.24
+    $ bin/deploy -b $GIT_REL --es-ip $ES_IP --cluster-name "$ENCD_REL" --remote-indexing --primary-indexing --index-procs 8 --app-workers 12 --pg-open --do-batchupgrade no --dry-run
```
Comment (hitz) [inline README-offloaded-indexing.md:12]: I think --primary-indexing is redundant here?   It appears to be on by default, unless you specify --no-indexing?
Also --do-batchupgrade no is set by --release-candidate or --candidate, right?

File: README-offloaded-indexing.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    set remote_indexing = true
+    set remote_indexing_threshold = 10001
+
+    # Check apache
+    App workers should be 12
+    indexer, vis, and region should exist
+
+    # Indexer node: 8 app workers, 16 index workers
+    $ export PG_IP=172.31.30.133
+    $ bin/deploy -b $GIT_REL --es-ip $ES_IP --pg-ip $PG_IP -n "$ENCD_REL-indexer" --cluster-name "$ENCD_REL" --no-indexing --primary-indexing --app-workers 8 --do-batchupgrade no --dry-run
```
Comment (hitz) [inline README-offloaded-indexing.md:38]: --no-indexing and --primary-indexing seem in opposition.    I think --primary-indexing overrides (from looking at deploy)...  probably no flags here are needed?
(also same comment about --do-batchupgrade)

File: README-offloaded-indexing.md (hunk 2 above)
Comment (hitz) [reply README-offloaded-indexing.md:37]: must be.  Does the previous command list the IP or do we have to pull it off the front end.

File: README-offloaded-indexing.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    App workers should be 8
+    indexer should exist
+    No vis and region
+
+    # turn off machine so we can check that the head node can turn it on for initial indexing
+        * wait for it to stop
+
+    # SSh onto head node and restart apache while watching the error.log
+        * wait for the head node to trigger remote indexing
+        * make sure remote indexer turns on in AWS
```
Comment (hitz) [inline README-offloaded-indexing.md:70]: oh the lambda function turns it on.   Do we have the lambda and or SQS code in github somewhere?

File: README-offloaded-indexing.md (hunk 3 above)
Comment (caseylitton) [reply README-offloaded-indexing.md:12]: I've been mentioning this in various configuration management discussions.   The answer is unfortunately no. 
 The dependencies between arguments in the config are not straightforward. 

We really need to clean up the configuration management before getting too much further into our scaling endeavors.

File: README-offloaded-indexing.md (hunk 2 above)
Comment (caseylitton) [reply README-offloaded-indexing.md:37]: yes, That is a good thing to note here.

File: README-offloaded-indexing.md (hunk 4 above)
Comment (caseylitton) [reply README-offloaded-indexing.md:38]: That could be true, but the vars are used for various parts of deployment.

I know these worked for me.  :)   And need people to deploy clusters with the commands we'll be using for v104 release.

File: README-offloaded-indexing.md (hunk 5 above)
Comment (caseylitton) [reply README-offloaded-indexing.md:70]: yes, Once we finish these cluster deployment tests we can look into that.

File: README-offloaded-indexing.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
+### Cluster with remote indexing node, this a little manual since deploy vars are getting messy
+    $ export GIT_REL='v103rc2' && export ENCD_REL='v103rc2' && echo "$GIT_REL $ENCD_REL"
+
+    # Normal elasticsearch cluster
+    $ bin/deploy -b "$GIT_REL" --cluster-name "$ENCD_REL" --es-wait --dry-run
```
Comment (hitz) [inline README-offloaded-indexing.md:5]: should not be --dry-run

**Comment by caseylitton** [comment]
moved to devops

**Pull request closed without merging**